.github/sync-repo-settings.yaml

```diff
@@ -0,0 +1,13 @@
+rebaseMergeAllowed: true
```

Comment:
Is there a source of truth for this file? If so, can you comment?

Reply:
Yes!  The config format for it is here:
https://github.com/googleapis/repo-automation-bots/tree/master/packages/sync-repo-settings#using-the-bot

This is a bot that verifies/sets these settings every night.  It looks like branch protection was supposed to be enabled in https://github.com/googleapis/google-cloudevents-go/issues/4, but all of these repos were showing up in our scans as out of compliance, and it's turned off again.  This config should help us keep it configured properly :)